src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs

```diff
@@ -609,7 +609,14 @@ void visitFunctionPointerSignature(IMethodSymbol symbol)
 
                 foreach (var param in symbol.Parameters)
                 {
-                    param.Accept(this.NotFirstVisitor);
+                    AddParameterRefKindIfRequired(param.RefKind);
```

Comment:
>AddParameterRefKindIfRequired [](start = 20, length = 29)

This still depends on some parameter display options, it feels like we don't want to depend on any of them.

Reply:
Always show ref kinds?

Reply:
> Always show ref kinds?

Probably use the same option that is used for return type.

---
In reply to: [578556365](https://github.com/dotnet/roslyn/pull/51223#discussion_r578556365) [](ancestors = 578556365)

Reply:
Return type checks for SymbolDisplayMemberOptions.IncludeRef. What I'm currently doing is consistent with that.

Reply:
Unless you want to use SymbolDisplayMemberOptions.IncludeRef for both parameters and return types? I'm not sure how much sense that makes.

Reply:
> Unless you want to use SymbolDisplayMemberOptions.IncludeRef for both parameters and return types? I'm not sure how much sense that makes.

What are you proposing? I think that the goal for this PR is to remove dependency on parameter display options when we display a function pointer type. Usage of ```AddParameterRefKindIfRequired``` simply doesn't get us there. Aligning behavior with the return type handling at least doesn't make things worse than they already are (around the return type).

---
In reply to: [579010475](https://github.com/dotnet/roslyn/pull/51223#discussion_r579010475) [](ancestors = 579010475)